src/pilot/assign_role.py

```diff
@@ -846,7 +846,11 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index,
     if role_index.index:
         role = "node:{}-{}".format(flavor, role_index.index)
 
-    value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
+    if 'capabilities' in node.properties:
+        value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
```

Comment:
What happens if `boot_mode` and/or `boot_option` are already in `capabilities`? It seems to me they should not be added.

This could occur when OOB inspection or IB introspection is performed before `assign_role.py`. Our `idrac` driver's OOB inspection, both WS-Man and Redfish, add `boot_mode` to `capabilities` if it is not already present and IB introspection can be configured to do the same, although its default setting is not to do that.

Reply:
I believe this will overwrite it as `boot_mode:uefi` and `boot_option:local`